### PR TITLE
Fix generator error handling

### PR DESCRIPTION
**Pull request by Kapin**

### CODE REVIEW

File: src-backend/api/generator.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def __init__(self, page):
         self.page = page
 
         if not self.page.elements.all():
-            raise ValueError('Page with display index %d is empty' % self.page.display_index)
+            raise ValueError('Page %d is empty' % self.page.display_index)
```
Comment (Trinovantes) [inline src-backend/api/generator.py]: leave the "display index" in

File: src-backend/api/generator.py [hunk 2]
```diff
@@ -54,22 +54,22 @@ def __init__(self, element):
         self.element = element
 
         if not self.element.element_type:
-            self.__raise_error('Element with id %d has no type')
+            self.__raise_error('Element on page %d has no type')
```
Comment (Trinovantes) [inline src-backend/api/generator.py]: Make this more verbose and say `Error on page id:% d:%`

The more debug info the better

**Comment by Ommy** [comment]
+1 after comments

**Comment by Trinovantes** [comment]
We really need to ban commits whose message are just "Code review comments". It makes it really hard to navigate through the logs

**Comment by Ommy** [comment]
Yea, rebase before committing this.

File: src-backend/api/generator.py [hunk 3]
```diff
@@ -54,22 +54,22 @@ def __init__(self, element):
         self.element = element
 
         if not self.element.element_type:
-            self.__raise_error('Element with id %d has no type')
+            self.__raise_error('Error on Page with display index %d: Element has no type')
```
Comment (Trinovantes) [inline src-backend/api/generator.py]: include the primary key too:
`Error on Page {primary_key} with display index {display_index}: {reason}`

**Comment by Trinovantes** [comment]
+1

**Comment by ConnorCimowsky** [comment]
LGTM

**Pull request merged**